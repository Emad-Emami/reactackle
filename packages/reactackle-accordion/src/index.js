import Accordion from './Accordion';
import AccordionItem from './AccordionItem/AccordionItem';

export { Accordion, AccordionItem };
