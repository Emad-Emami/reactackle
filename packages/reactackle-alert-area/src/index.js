import AlertArea from './AlertArea';

export { AlertArea };
