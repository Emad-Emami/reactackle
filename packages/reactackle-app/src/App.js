'use strict';

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  createBroadcast,
  APP_CHANNEL,
  registerDefaultComponentTheme,
} from 'reactackle-core';
import { AppStyled } from './styles/AppStyled';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('app', componentTheme);

const propTypes = {
  /**
   * Set App height to be equal to the window height, so the <Content /> will
   * become scrollable.
   */
  fixed: PropTypes.bool,
};

const defaultProps = {
  fixed: false,
};

const childContextTypes = {
  [APP_CHANNEL]: PropTypes.func.isRequired,
};

export default class App extends Component {
  constructor(props) {
    super(props);

    this._broadcast = createBroadcast({
      fixed: this.props.fixed,
    });
  }

  getChildContext() {
    return {
      ...this.context,
      [APP_CHANNEL]: this._broadcast.subscribe,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.fixed !== nextProps.fixed) {
      this._broadcast.publish({
        fixed: nextProps.fixed,
      });
    }
  }

  render() {
    return (
      <AppStyled fixed={this.props.fixed}>
        {this.props.children}
      </AppStyled>
    );
  }
}

App.childContextTypes = childContextTypes;
App.propTypes = propTypes;
App.defaultProps = defaultProps;
App.displayName = 'App';
