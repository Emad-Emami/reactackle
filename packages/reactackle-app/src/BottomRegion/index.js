import BottomRegion from './BottomRegion';

export { BottomRegion };
