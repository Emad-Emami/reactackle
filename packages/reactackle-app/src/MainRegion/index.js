import MainRegion from './MainRegion';

export { MainRegion };
export * from './Content';
