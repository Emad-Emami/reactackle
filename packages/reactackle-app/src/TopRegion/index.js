import TopRegion from './TopRegion';

export { TopRegion };
