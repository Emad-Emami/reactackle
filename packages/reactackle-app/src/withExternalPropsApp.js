import { APP_CHANNEL, withExternalProps } from 'reactackle-core';

export default withExternalProps(APP_CHANNEL);
