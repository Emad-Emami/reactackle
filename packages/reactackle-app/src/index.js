import App from './App';

export { App };
export * from './BottomRegion';
export * from './TopRegion';
export * from './MainRegion';
