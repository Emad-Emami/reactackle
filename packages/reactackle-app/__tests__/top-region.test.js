'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { TopRegion } from '../src';

jest.mock('react-dom');

describe('<TopRegion/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <TopRegion />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop fixed set true', () => {
    const tree = renderer.create(
      <TopRegion fixed />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop fixed set false', () => {
    const tree = renderer.create(
      <TopRegion fixed={false} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if contains children', () => {
    const tree = renderer.create(
      <TopRegion>
        <div>TEST</div>
      </TopRegion>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
