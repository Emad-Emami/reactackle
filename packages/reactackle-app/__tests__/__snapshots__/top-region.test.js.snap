// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopRegion/> renders correctly if contains children 1`] = `
<div
  className="TopRegionStyled-cVQlko gRcViP"
>
  <div
    className="TopRegionContentStyled-iFwSwY ewMYrE"
  >
    <div>
      TEST
    </div>
  </div>
</div>
`;

exports[`<TopRegion/> renders correctly if prop fixed set false 1`] = `
<div
  className="TopRegionStyled-cVQlko gRcViP"
>
  <div
    className="TopRegionContentStyled-iFwSwY ewMYrE"
  />
</div>
`;

exports[`<TopRegion/> renders correctly if prop fixed set true 1`] = `
<div
  className="TopRegionStyled-cVQlko gRcViP"
>
  <div
    className="TopRegionContentStyled-iFwSwY ewMYrE"
  />
</div>
`;

exports[`<TopRegion/> renders correctly with default props 1`] = `
<div
  className="TopRegionStyled-cVQlko gRcViP"
>
  <div
    className="TopRegionContentStyled-iFwSwY ewMYrE"
  />
</div>
`;
